opencog/guile/SchemeSmob.cc

```diff
@@ -36,7 +36,7 @@ using namespace opencog;
  * simplify that code, and probably improve performance just a bit.
  */
 
-scm_t_bits SchemeSmob::cog_misc_tag;
+std::atomic<scm_t_bits> SchemeSmob::cog_misc_tag;
```

Comment:
This is a nonsense fix.  `cog_misc_tag` is set once and only once ever, making it atomic cannot possibly solve any problem that you are having.